runtime/src/append_vec.rs

```diff
@@ -523,4 +552,145 @@ pub mod tests {
+            let data_len: &mut u64 = unsafe { &mut *(data_len as *const u64 as *mut u64) };
+            *data_len = too_large_data_len;
+        }
+
+        assert_eq!(account.meta.data_len, too_large_data_len);
+        let accounts = av.accounts(0);
+        assert_matches!(accounts.first(), None);
+
+        av.flush().unwrap();
+        let result = av.set_file(path);
+        assert_matches!(result, Err(ref message) if message.to_string() == *"incorrect layout/length");
+    }
+
+    #[test]
+    fn test_set_file_crafted_executable() {
+        let file = get_append_vec_path("test_set_file_crafted_executable");
+        let path = &file.path;
+        let mut av = AppendVec::new(&path, true, 1024 * 1024);
+        av.append_account_test(&create_test_account(10)).unwrap();
+        {
+            let mut executable_account = create_test_account(10);
+            executable_account.1.executable = true;
+            av.append_account_test(&executable_account).unwrap();
+        }
+
+        // reload accounts
+        let accounts = av.accounts(0);
+        let account = &accounts[0];
+
+        // ensure false is 0u8 and true is 1u8 actually
+        {
+            let executable_bool: &bool = &account.account_meta.executable;
+            // UNSAFE: Force to interpret mmap-backed bool as u8 to really read the actual memory content
+            let executable_byte: &u8 = unsafe { &*(executable_bool as *const bool as *const u8) };
+            assert_eq!(*executable_byte, 0);
+
+            let executable_bool: &bool = &accounts[1].account_meta.executable;
+            // UNSAFE: Force to interpret mmap-backed bool as u8 to really read the actual memory content
+            let executable_byte: &u8 = unsafe { &*(executable_bool as *const bool as *const u8) };
+            assert_eq!(*executable_byte, 1);
+        }
+
+        let crafted_executable = u8::max_value() - 1;
+
+        {
+            let executable_ref: &bool = &account.account_meta.executable;
+            #[allow(mutable_transmutes)]
+            // UNSAFE: Force to interpret mmap-backed &bool as &u8 to write some crafted value;
+            let executable_byte: &mut u8 =
+                unsafe { &mut *(executable_ref as *const bool as *mut u8) };
+            *executable_byte = crafted_executable;
+        }
+
+        // reload crafted accounts
+        let accounts = av.accounts(0);
+        let account = accounts.first().unwrap();
+
+        // we can observe crafted value by ref
+        {
+            let executable_bool: &bool = &account.account_meta.executable;
+            // we can not use assert_eq!...
+            // *executable_bool is true but its actual memory value is crafted_executable, not 1
+            assert!(*executable_bool != true);
+            // UNSAFE: Force to interpret mmap-backed bool as u8 to really read the actual memory content
+            let executable_byte: &u8 = unsafe { &*(executable_bool as *const bool as *const u8) };
```

Comment:
this unsafe block/casting is repeated in the tests a few times, can we have a function that is `assert_eq_bool(ptr, expected_bool_value);`

Reply:
Yeah, I was a bit annoyed the repeated `unsafe`s... Thanks for suggestion! I've done the cleanup differentially, though. How does that look for you?: 6d62daab356f2535746d88043fff199ee8c1e827